Mathlib/Util/Qq.lean

```diff
@@ -40,4 +40,21 @@ def findLocalDeclWithTypeQ? {u : Level} (sort : Q(Sort u)) : MetaM (Option Q($so
 This is a Qq version of `Lean.Meta.mkDecideProof`. -/
 def mkDecideProofQ (p : Q(Prop)) : MetaM Q($p) := mkDecideProof p
 
+/-- Join a list of elements of type `α` into a container `β`.
+
+Usually `β` is `q(Multiset α)` or `q(Finset α)` or `q(Set α)`.
+Note: this is needed since there aren't `ToExpr` instances on these types, which is to be expected,
+since, e.g, there is in general no canonical way of ordering the elements in a Set/Finset/Multiset.
```

Comment:
This is a bad reason, because you could just use `unquot` and everything would be fine!

The problem for `Finset` is the decidability instance:

https://github.com/leanprover-community/mathlib4/blob/b7487412d3ed47bfc2786256b301b1cbd07d6bdc/MathlibTest/eval_elab.lean#L23-L47

Reply:
For `Set`, the explanation is "there is obviously no way to enumerate in any order'.

For `Multiset`, I think there's no excuse and we absolutely should have the `ToExpr` isntance.

I think a better argument here is that you might want to build a finset literal from things that are not compile-time constants; the benefit of the definition in this PR is that the signature lets the caller produce the individual terms via mechanisms other than recursively calling `toExpr`.

Reply:
The original motivation for this was to build a finset literal from things which are not compile-time constants!

Reply:
I'm pretty sure `unsafe toExpr x` works just fine in a `simproc`?

Reply:
> that the signature lets the caller produce the individual terms via mechanisms other than recursively calling `toExpr`.

To elaborate on this a little; it means it can be used to produce sets of `Real` number multiples of `pi`.

Reply:
Let's discuss this elsewhere!

Reply:
Discussed, Eric is right :)